compiler/semexprs.nim

```diff
@@ -662,6 +662,8 @@ proc semArrayConstr(c: PContext, n: PNode, flags: TExprFlags; expectedType: PTyp
       #result.add fitNode(c, typ, n[i])
       inc(lastIndex)
     addSonSkipIntLit(result.typ, typ, c.idgen)
+    if typ.skipTypes({tyPtr, tyRef}).kind in {tyGenericInst, tyObject}:
```

Comment:
An object is a concrete nominal type and a tyGenericInst is an instance of a generic, not necessarily a generic from an object so this check cannot possibly correct, no matter why it exists.

Reply:
here is just ensure it gonna compare object types in `typeRel`, maybe should use some like `skipToObject`.